src/lib/layer-manager.js

```diff
@@ -147,29 +147,38 @@ export default class LayerManager {
   }
 
   pickLayer({x, y, mode, radius = 0}) {
-    const {gl, uniforms} = this.context;
+    const {gl} = this.context;
 
-    // Set up a frame buffer if needed
-    if (this.context.pickingFBO === null ||
-      gl.canvas.width !== this.context.pickingFBO.width ||
-      gl.canvas.height !== this.context.pickingFBO.height) {
-      this.context.pickingFBO = new FramebufferObject(gl, {
-        width: gl.canvas.width,
-        height: gl.canvas.height
-      });
-    }
     return pickLayers(gl, {
       x,
       y,
       radius,
       layers: this.layers,
       mode,
       viewport: this.context.viewport,
-      pickingFBO: this.context.pickingFBO,
+      pickingFBO: this._getPickingBuffer(),
       lastPickedInfo: this.context.lastPickedInfo
     });
   }
 
+  queryLayer({x, y, width, height}, layerIds) {
```

Comment:
Maybe comments above this and pickLayer so that the reader understands how and why they are different